tests/src/Type/data/inspector.php

```diff
@@ -3,155 +3,206 @@
+    Inspector::assertAllNotEmpty($mixed_arg)
+        ? assertType('iterable<float|int<min, -1>|int<1, max>|object|resource|non-empty-string|non-empty-array>', $mixed_arg)
+        : assertType('mixed~iterable<float|int<min, -1>|int<1, max>|object|resource|non-empty-string|non-empty-array>', $mixed_arg);
+
+    Inspector::assertAllNotEmpty($array_arg)
+        ? assertType('array<float|int<min, -1>|int<1, max>|object|resource|non-empty-string|non-empty-array>', $array_arg)
+        : assertType('array', $array_arg);
+}
+
+/**
+ * @see Inspector::assertAllNumeric()
+ */
+function assertAllNumeric(mixed $mixed_arg, array $array_arg): void {
+    Inspector::assertAllNumeric($mixed_arg)
+        ? assertType('iterable<float|int>', $mixed_arg)
+        : assertType('mixed~iterable<float|int>', $mixed_arg);
+
+    Inspector::assertAllNumeric($array_arg)
+        ? assertType('array<float|int>', $array_arg)
+        : assertType('array', $array_arg);
+}
+
+/**
+ * @see Inspector::assertAllMatch()
+ */
+function assertAllMatch(mixed $mixed_arg, array $array_arg): void {
+    $pattern = 'foo';
+    Inspector::assertAllMatch($pattern, $mixed_arg, false)
+        ? assertType('iterable<string>', $mixed_arg)
+        : assertType('mixed~iterable<string>', $mixed_arg);
+
+    Inspector::assertAllMatch($pattern, $array_arg, false)
+        ? assertType('array<string>', $array_arg)
+        : assertType('array', $array_arg);
+}
+
+/**
+ * @see Inspector::assertAllRegularExpressionMatch()
+ */
+function assertAllRegularExpressionMatch(mixed $mixed_arg, array $array_arg): void {
+    $pattern = 'foo';
+    Inspector::assertAllRegularExpressionMatch($pattern, $mixed_arg)
+        ? assertType('iterable<string>', $mixed_arg)
+        : assertType('mixed~iterable<string>', $mixed_arg);
+
+    Inspector::assertAllRegularExpressionMatch($pattern, $array_arg)
+        ? assertType('array<string>', $array_arg)
+        : assertType('array', $array_arg);
+}
+
+/**
+ * @see Inspector::assertAllObjects()
+ */
+function assertAllObjects(mixed $mixed_arg, array $array_arg): void {
+    // Note: 'TranslatableMarkup' is not mentioned in the final union type,
+    // because it is a subtype of '\Stringable'.
+    Inspector::assertAllObjects($mixed_arg, Node::class, TranslatableMarkup::class, '\\Stringable', '\\Drupal\\jsonapi\\JsonApiResource\\ResourceIdentifier')
+        ? assertType('iterable<\Drupal\jsonapi\JsonApiResource\ResourceIdentifier|Drupal\node\Entity\Node|\Stringable>', $mixed_arg)
+        : assertType('mixed~iterable<\Drupal\jsonapi\JsonApiResource\ResourceIdentifier|Drupal\node\Entity\Node|\Stringable>', $mixed_arg);
+
+    Inspector::assertAllObjects($array_arg, Node::class, TranslatableMarkup::class, '\\Stringable', '\\Drupal\\jsonapi\\JsonApiResource\\ResourceIdentifier')
+        ? assertType('array<\Drupal\jsonapi\JsonApiResource\ResourceIdentifier|Drupal\node\Entity\Node|\Stringable>', $array_arg)
+        : assertType('array', $array_arg);
```

Comment:
Also, this test case is a little bit different because it now includes `Node::class` as well. Previously (now as well), `TranslatableMarkup::class` was expanded by `\Stringable`, so there were actually no test cases for that type of class-string passing. `Node::class` is covering it now.

Previously, it looked like `TranslatableMarkup::class` was completely ignored.